File: drills/ts/harnesses/nc116_connected_components.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.countComponents !== "function") throw new Error("__signature_mismatch__");
  return [
    ["countComponents(5, [[0,1],[1,2],[3,4]])", show(2), show(solution.countComponents(5, [[0, 1], [1, 2], [3, 4]]))],
    ["countComponents(5, [[0,1],[1,2],[2,3],[3,4]])", show(1), show(solution.countComponents(5, [[0, 1], [1, 2], [2, 3], [3, 4]]))],
    ["countComponents(3, [])", show(3), show(solution.countComponents(3, []))],
    ["countComponents(0, [])", show(0), show(solution.countComponents(0, []))],
    ["countComponents(1, [])", show(1), show(solution.countComponents(1, []))],
    ["countComponents(4, [[0,1],[1,0]])", show(3), show(solution.countComponents(4, [[0, 1], [1, 0]]))],
  ];
}

File: drills/elixir/harnesses/nc116_connected_components.exs
[
  {"count_components(5, [[0, 1], [1, 2], [3, 4]])", inspect(2),
   inspect(Solution.count_components(5, [[0, 1], [1, 2], [3, 4]]))},
  {"count_components(5, [[0, 1], [1, 2], [2, 3], [3, 4]])", inspect(1),
   inspect(Solution.count_components(5, [[0, 1], [1, 2], [2, 3], [3, 4]]))},
  {"count_components(3, [])", inspect(3), inspect(Solution.count_components(3, []))},
  {"count_components(0, [])", inspect(0), inspect(Solution.count_components(0, []))},
  {"count_components(1, [])", inspect(1), inspect(Solution.count_components(1, []))},
  {"count_components(4, [[0, 1], [1, 0]])", inspect(3),
   inspect(Solution.count_components(4, [[0, 1], [1, 0]]))}
]
